packages/rum-core/src/domain/contexts/pageStateContexts.ts

```diff
@@ -0,0 +1,122 @@
+import type { Duration, RelativeTime } from '@datadog/browser-core'
+import {
+  isExperimentalFeatureEnabled,
+  noop,
+  addDuration,
+  addEventListeners,
+  relativeNow,
+  DOM_EVENT,
+} from '@datadog/browser-core'
+
+export const PAGE_STATE_CONTEXT_MAX_LENGTH = 500
+
+export const enum PageState {
+  ACTIVE = 'active',
+  PASSIVE = 'passive',
+  HIDDEN = 'hidden',
+  FROZEN = 'frozen',
+  TERMINATED = 'terminated',
+}
+
+export type PageStateContext = Array<{ state: PageState; startTime: RelativeTime }>
+
+export interface PageStateContexts {
+  getPageStates: (startTime: RelativeTime, duration: Duration) => PageStateContext | undefined
+  stop: () => void
+}
+
+let pageStateContext: PageStateContext = []
+let state: PageState | undefined
+
+export function startPageStateContexts(): PageStateContexts {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏The behaviour of this component seems quite similar to ContextHistory, if we need to keep it after the investigation, it could be interesting to mutualize some code.
In the mean time, it could be interesting to take inspiration from the naming we have there to be a bit clearer and consistent.
Something like:
```suggestion
export const enum PageState {
  ACTIVE = 'active',
  PASSIVE = 'passive',
  HIDDEN = 'hidden',
  FROZEN = 'frozen',
  TERMINATED = 'terminated',
}

export type PageStateEntry = { pageState: PageState; startTime: RelativeTime }

export interface PageStateHistory {
  findAll: (startTime: RelativeTime, duration: Duration) => PageStateEntry[] | undefined
  stop: () => void
}

let pageStateEntries: PageStateEntry[] = []
let pageState: PageState | undefined

export function startPageStateHistory(): PageStateHistory {
```